src/control/cmd/agent/mgmt_rpc.go

```diff
@@ -63,14 +83,36 @@ func (mod *mgmtModule) ID() int32 {
 	return drpc.ModuleMgmt
 }
 
-func (mod *mgmtModule) handleGetAttachInfo(reqb []byte) ([]byte, error) {
-	if os.Getenv("DAOS_AGENT_DISABLE_CACHE") == "true" {
-		mod.cachedAttachInfo = false
-		mod.log.Debugf("GetAttachInfo agent caching has been disabled\n")
+// handleGetAttachInfo invokes the GetAttachInfo dRPC.  The agent determines the
+// NUMA node for the client process based on its PID.  Then based on the
+// server's provider, chooses a matching network interface and domain from the
+// client machine that has the same NUMA affinity.  It is considered an error if
+// the client application is bound to a NUMA node that does not have a network
+// device / provider combination with the same NUMA affinity.
+//
+// The client machine may have more than one matching network interface per
+// NUMA node.  In order to load balance the client application's use of multiple
+// network interfaces on a given NUMA node, a round robin resource allocation
+// scheme is used to choose the next device for that node.  See attachInfoCache
+// "loadBalance()" for details.
```

Comment:
A little hesitant about a comment with this much detail being so far from its actual implementation. Seems doomed to fall out of date, and I'm not sure the assumption is really important to understand what's happening in this function. I feel like the overall explanation belongs in the README.md for the agent. You should definitely update the README to document the cache.

Reply:
I thought the comments were helpful to explain what was happening.  I've removed the section you highlighted and will look at adding it elsewhere in a doc update.